WordPress/src/main/java/org/xmlrpc/android/ApiHelper.java

```diff
@@ -1190,4 +1014,54 @@ else if (attrName.equals("href"))
         }
         return null; // never found the rsd tag
     }
+
+    /*
+     * fetches a single post saves it to the db - note that this should NOT be called from main thread
+     */
+    public static boolean updateSinglePost(int localBlogId, String remotePostId, boolean isPage) {
+        Blog blog = WordPress.getBlog(localBlogId);
+        if (blog == null || TextUtils.isEmpty(remotePostId)) {
+            return false;
+        }
+
+        XMLRPCClientInterface client = XMLRPCFactory.instantiate(
+                blog.getUri(),
+                blog.getHttpuser(),
+                blog.getHttppassword());
+
+        Object[] apiParams;
+        if (isPage) {
+            apiParams = new Object[]{
+                    blog.getRemoteBlogId(),
+                    remotePostId,
+                    blog.getUsername(),
+                    blog.getPassword()
+            };
+        } else {
+            apiParams = new Object[]{
+                    remotePostId,
+                    blog.getUsername(),
+                    blog.getPassword()
+            };
+        }
+
+        try {
+            Object result = client.call(isPage ? "wp.getPage" : "metaWeblog.getPost", apiParams);
+
+            if (result != null && result instanceof Map) {
+                Map postMap = (HashMap) result;
+                List<Map<?, ?>> postsList = new ArrayList<>();
+                postsList.add(postMap);
+
+                WordPress.wpDB.savePosts(postsList, localBlogId, isPage, true);
```

Comment:
Something's not quite working with the db `update` call in `savePosts`. From a quick debug I didn't see any problems with the WHERE clause, but it ends up with two rows for the Post for some reason:

<img width="621" alt="screen shot 2015-07-24 at 4 42 56 pm" src="https://cloud.githubusercontent.com/assets/789137/8886766/ea179b02-3224-11e5-876f-5c90d7a738d5.png">